velox/expression/CastExpr-inl.h

```diff
@@ -293,6 +293,47 @@ VectorPtr CastExpr::applyDecimalToBooleanCast(
   return result;
 }
 
+template <typename FromNativeType>
+VectorPtr CastExpr::applyDecimalToVarcharCast(
+    const SelectivityVector& rows,
+    const BaseVector& input,
+    exec::EvalCtx& context,
+    const TypePtr& fromType,
+    const TypePtr& toType) {
+  VectorPtr result;
+  context.ensureWritable(rows, VARCHAR(), result);
+  (*result).clearNulls(rows);
+  const auto simpleInput = input.as<SimpleVector<FromNativeType>>();
+  auto flatResult = result->asFlatVector<StringView>();
+
+  // Calculate the total size of the stringBuffers.
+  size_t totalResultBytes = 0;
+  applyToSelectedNoThrowLocal(context, rows, result, [&](int row) {
+    if (!simpleInput->isNullAt(row)) {
+      std::string output =
+          DecimalUtil::toString((int128_t)simpleInput->valueAt(row), fromType);
+      totalResultBytes += output.size();
+    }
+  });
+
+  auto rawBuffer = flatResult->getRawStringBufferWithSpace(totalResultBytes);
```

Comment:
Only the not inline StringView need to have a reference in `std::vector<BufferPtr>&& stringBuffers,`

Reply:
This part was removed because it can be handled with existing method at [FlatVector.cpp#L263-L277](https://github.com/facebookincubator/velox/blob/main/velox/vector/FlatVector.cpp#L263-L277).